test/e2e/walkthrough.go

```diff
@@ -142,6 +143,15 @@ var _ = framework.ServiceCatalogDescribe("walkthrough", func() {
 		)
 		Expect(err).NotTo(HaveOccurred(), "failed to wait instance to be ready")
 
+		// Make sure references have been resolved
```

Comment:
I skimmed `controller_instance_test.go` and didn't see an analogue there; we should also add a unit test for the references being resolved. The setup for the e2e tests is much more involved and I suspect most developers just wait for Jenkins to run them to see if they pass, while they're likely running the unit and integration tests locally during development.

Reply:
Yes, there are tests in clientset_test that validates these. In any case, good idea, added some here.